File: Android-Demo/IPC/IPCMaster/app/src/main/java/com/wx/ipcmaster/MainActivity.java
package com.wx.ipcmaster;

import android.databinding.ViewDataBinding;
import android.view.View;

import com.wx.ipcmaster.activity.AidlActivity;
import com.wx.ipcmaster.activity.MessengerActivity;
import com.wx.ipcmaster.activity.base.BaseActivity;
import com.wx.ipcmaster.databinding.ActivityMainBinding;

public class MainActivity extends BaseActivity {

    private ActivityMainBinding binding;

    @Override
    public int getLayoutId() {
        return R.layout.activity_main;
    }

    @Override
    public void setDataBinding(ViewDataBinding binding) {
        this.binding = (ActivityMainBinding) binding;
    }

    @Override
    public void initData() {
        binding.btn1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                openActivity(MessengerActivity.class);
            }
        });

        binding.btn2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                openActivity(AidlActivity.class);
            }
        });

    }

}

File: Android-Demo/ThreadMaster/app/src/main/java/com/wx/threadmaster/MainActivity.kt
package com.wx.threadmaster

import android.databinding.ViewDataBinding
import com.wx.glidedemo1.base.BaseActivity
import com.wx.threadmaster.activity.BasicUsageActivity
import com.wx.threadmaster.databinding.ActivityMainBinding

class MainActivity : BaseActivity() {

    private lateinit var binding: ActivityMainBinding

    override val layoutId: Int
        get() = R.layout.activity_main

    override fun setDataBinding(binding: ViewDataBinding) {
        this.binding = binding as ActivityMainBinding
    }

    override fun initData() {
        binding.btn1.setOnClickListener {
            openActivity(BasicUsageActivity::class.java)
        }
    }

}
